### PR TITLE
Fixed int.real, int.numerator etc. returns wrong value when it's int type.

### LINKED ISSUE #3687: int.real returns self for `int` type without subclassing
when a is int,
```python
assert a.real is a
```

when a is subclass of int,
```python
assert a.real is not a
assert type(a.real) is int
```

### PR DESCRIPTION
**Pull request by killme2008**
Try to fix #3687

### CODE REVIEW

**Comment by Snowapril** [comment]
Test issue3687 must be in extra_tests/snippets/builtin_int.py

**Comment by killme2008** [comment]
@Snowapril Got it, any other comments?

**Review by Snowapril — APPROVED** [review]
Looks good to me 😊

File: vm/src/builtins/int.rs [hunk 1]
```diff
@@ -688,10 +688,18 @@ impl PyInt {
...
     }
 
+    #[inline]
+    fn long_long(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyRef<Self> {
+        if zelf.class().is(&vm.ctx.types.int_type) {
+            return zelf;
+        }
+
+        vm.ctx.new_bigint(&zelf.value)
+    }
```
Comment (Snowapril) [inline vm/src/builtins/int.rs:698]: Maybe it is trivial, below one seems better
```rust
if zelf.class().is(&vm.ctx.types.int_type) {
    zelf
} else {
    vm.ctx.new_bigint(&zelf.value)
}
```

File: vm/src/builtins/int.rs [hunk 2]
```diff
@@ -688,10 +688,18 @@ impl PyInt {
         Ok(bytes.into())
     }
 
+    #[inline]
+    fn long_long(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyRef<Self> {
+        if zelf.class().is(&vm.ctx.types.int_type) {
+            return zelf;
+        }
+
+        vm.ctx.new_bigint(&zelf.value)
```
Comment (Snowapril) [inline vm/src/builtins/int.rs:697]:
```suggestion
        if zelf.class().is(&vm.ctx.types.int_type) {
            zelf
        } else {
            vm.ctx.new_bigint(&zelf.value)
        }
```

**Review by youknowone — CHANGES_REQUESTED** [review]
looks good! thank you!
I left a minor comment about the name. could you please check it?

@Snowapril thank you for the review!

File: vm/src/builtins/int.rs [hunk 3]
```diff
@@ -688,10 +688,18 @@ impl PyInt {
         Ok(bytes.into())
     }
 
+    #[inline]
+    fn long_long(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyRef<Self> {
```
[the remark this answers is not in the document]
Comment (killme2008) [reply vm/src/builtins/int.rs:692]: Yep, `clone_if_subclass` is a better name. `long_long` is from cpython.

**Review by youknowone — APPROVED** [review]
looks good, thank you!

**Pull request merged**